lib/create-subdomain/service.ts

```diff
@@ -167,12 +154,18 @@ export const createSubdomain = async ({
 
       await clientWithWallet.addChain({ chain: selectedChain });
 
-      const { request } = await clientWithWallet.simulateContract({
-        ...calldata,
-        address: contractAddress,
-      });
-
-      await clientWithWallet.writeContract(request);
+      try {
+        const { request } = await clientWithWallet.simulateContract({
+          ...calldata,
+          address: contractAddress,
+        });
+
+        await clientWithWallet.writeContract(request);
+      } catch (error: any) {
+        toast.error(error?.cause?.reason ?? "Error creating subdomain");
+        await clientWithWallet.switchChain({ id: chains.sepolia.id });
```

Comment:
I recommend reviewing if necessarily this is a chain switch error, if not, improve this

Reply:
@FrancoAguzzi What do you mean by chain switch error?